### DISCUSSION TITLE
Display Modified Time of File on Status Bar?

### DISCUSSION THREAD

**Discussion by bertradio** [sxyazi/yazi#885] [1. Q&A]
Is there a way to do this?

**Comment by bertradio** [comment]
I was able to modify the status line by copying Status:render into init.lua in the yazi config directory. What variable would I use to display the modified time of the current file?

**Reply by sxyazi** [reply]
Use [`modified`](https://yazi-rs.github.io/docs/plugins/types#shared.cha), you can check out the linemode implementation:

https://github.com/sxyazi/yazi/blob/884de41b6606d805d9465004b5b3ce7758e065d8/yazi-plugin/preset/components/folder.lua#L19-L21

**Comment by bertradio** [comment]
I'm not a Rust programmer and not sure how to do this. The code I have is below. What would I need to add? I want the modified time to be before the percentage.

function Status:render(area)
self.area = area

local left = ui.Line { self:mode(), self:size(), self:name() }
local right = ui.Line { self:percentage(), self:position() }
return {
ui.Paragraph(area, { left }),
ui.Paragraph(area, { right }):align(ui.Paragraph.RIGHT),
table.unpack(Progress:render(area, right:width())),
}
end

**Reply by lpnh** [reply]
This code is written in Lua, not Rust. Add the following content to your `~/.config/yazi/init.lua` file, replacing any `Status:render` already present:

```
function Status:mtime()
	local h = cx.active.current.hovered
	if not h then
		return ui.Span("")
	end

	local time = h.cha.modified
	return ui.Span(time and os.date("%y-%m-%d %H:%M ", time // 1) or "")
end

function Status:render(area)
	self.area = area

	local left = ui.Line({ self:mode(), self:size(), self:name() })
	local right = ui.Line({ self:mtime(), self:percentage(), self:position() })
	return {
		ui.Paragraph(area, { left }),
		ui.Paragraph(area, { right }):align(ui.Paragraph.RIGHT),
		table.unpack(Progress:render(area, right:width())),
	}
end
```

While it's not perfect, it can serve as a starting point.

**Reply by bertradio** [reply]
Thanks so much. That's exactly what I needed. LOVE yazi!!
